testsuite/MDAnalysisTests/analysis/test_wbridge.py

```diff
@@ -99,5 +107,119 @@ def test_same_selection(self):
   1.0   1.0   1.0'''
         u = MDAnalysis.Universe(StringIO(grofile), format='gro')
         wb = WaterBridgeAnalysis(u, 'protein', 'protein')
-        wb.run()
+        wb.run(verbose=False)
         assert wb._timeseries == [[]]
+
+    def test_water_network(self):
+        '''
+        This test tests if the internal water object is generated correctly.
+        :return:
+        '''
+        grofile = '''Test gro file
+5
+    1ALA      N    1   0.000   0.000   0.000
+    1ALA      H    2   0.100   0.000   0.000
+    2SOL     OW    3   0.300   0.000   0.000
+    2SOL    HW2    4   0.400   0.000   0.000
+    4ALA      O    5   0.600   0.000   0.000
+ 1.0   1.0   1.0'''
+        u = MDAnalysis.Universe(StringIO(grofile), format='gro')
+        wb = WaterBridgeAnalysis(u, 'protein and (resid 1)', 'protein and (resid 4)')
+        wb.run(verbose=False)
+        water_network = wb._water_network[0]
+        assert list(water_network.keys()) == [('SOL', 2)]
+        values = list(water_network.values())
+        assert list(values[0][0])[0][:4] == (1, 2, ('ALA', 1, 'H'), ('SOL', 2, 'OW'))
+        assert list(values[0][1])[0][:4] == (3, 4, ('SOL', 2, 'HW2'), ('ALA', 4, 'O'))
+
+    def test_count_by_type_single_link(self):
+        '''
+        This test tests the simplest water bridge to see if count_by_type() works.
+        :return:
+        '''
+        grofile = '''Test gro file
+5
+    1ALA      N    1   0.000   0.000   0.000
+    1ALA      H    2   0.100   0.000   0.000
+    2SOL     OW    3   0.300   0.000   0.000
+    2SOL    HW2    4   0.400   0.000   0.000
+    4ALA      O    5   0.600   0.000   0.000
+ 1.0   1.0   1.0'''
+        u = MDAnalysis.Universe(StringIO(grofile), format='gro')
+        wb = WaterBridgeAnalysis(u, 'protein and (resid 1)', 'protein and (resid 4)')
+        wb.run(verbose=False)
+        assert wb.count_by_type().tolist() == [(1, 4, 'ALA', 1, 'H', 'ALA', 4, 'O',  1.)]
+
+    def test_count_by_type_multiple_link(self):
+        '''
+        This test tests if count_by_type() can assemble linkage from water network.
+        :return:
+        '''
+        grofile = '''Test gro file
+5
+    1ALA      N    1   0.000   0.000   0.000
+    1ALA      H    2   0.100   0.000   0.000
+    2SOL     OW    3   0.300   0.000   0.000
+    2SOL    HW2    4   0.400   0.000   0.000
+    4ALA      O    5   0.600   0.000   0.000
+ 1.0   1.0   1.0'''
+        u = MDAnalysis.Universe(StringIO(grofile), format='gro')
+        wb = WaterBridgeAnalysis(u, 'protein and (resid 1)', 'protein and (resid 4)')
+        # Build an dummy WaterBridgeAnalysis object for testing
+        wb._timeseries = True
+        wb.timesteps = range(1)
```

Comment:
Should this be a real list or array or is a range object sufficient?

Reply:
@orbeckst frequency = occurance / len(wb.timesteps)
So anything which allows len() to work is fine.